core/src/main/java/org/infinispan/interceptors/impl/TxInterceptor.java

```diff
@@ -604,6 +604,8 @@ private Object verifyRemoteTransaction(RemoteTxInvocationContext ctx, AbstractTr
             log.tracef("Rolling back remote transaction %s because either already completed (%s) or originator no longer in the cluster (%s).",
                        globalTransaction, alreadyCompleted, originatorMissing);
          }
+         // The rollback command only marks the transaction as completed in invokeAsync()
+         txTable.markTransactionCompleted(globalTransaction, false);
```

Comment:
Missed it, sorry. It would have been nice if I didn't, because I would have realized I can remove the `markTransactionCompleted()` call (at this point we know the transaction was already marked as completed) and some more dead code.

Now to the answer... `verifyRemoteTransaction` doesn't care if the lock command failed or not, only if the originator left the cluster (removed now) or if another thread already marked the transaction as completed (either because the originator left the cluster, or because the lock/prepare RPC timed out and the originator already sent out a rollback command).

`TxInterceptor` never marked the transaction for rollback anyway, only released the current command's locks, because of the `FAIL_SILENTLY` flag. IMO it would have been much better if the flag only applied to non-optimistic lock timeouts, but it applies to all exceptions, so it's `InvocationContextInterceptor` that marks the transaction for rollback (and only the TM transaction on the originator, not the cache transaction).